PxGet/Functions/Invoke-PxGet.ps1

```diff
@@ -0,0 +1,152 @@
+        $env:PSModulePath = Join-Path -Path (Get-Item $($moduleRoot)).Parent.FullName -ChildPath 'PSModules' -Resolve
+        Import-Module -Name 'PackageManagement'
+        Import-Module -Name 'PowerShellGet'
+
+        if( Test-Path -Path ($(Get-RootDirectory) + '\pxget.json') )
+        {
+            $pxModules = Get-Content -Path ($(Get-RootDirectory) + '\pxget.json') | ConvertFrom-Json
+        }
+        else 
+        {
+            Write-Error 'The pxget.json file does not exist!' -ErrorAction SilentlyContinue
+            return
+        }
+        
+        if( $pxModules )
+        {
+            $moduleNames = $pxModules.PSModules | Select-Object -ExpandProperty 'Name'
+            if( $moduleNames )
+            {
+                $modules = Find-Module -Name $moduleNames -ErrorAction Ignore
+
+                # We only care if the module is in PSModules right now. Later we'll allow dev dependencies, which can be
+                # installed globally.
+                $env:PSModulePath = $psModulesPath
+                foreach( $pxModule in $pxModules.PSModules )
+                {
+                    $allowPrerelease = $pxModule.Version.Contains('-')
+                    
+                    if( $allowPrerelease)
+                    {
+                        # Find module again but with AllVersions and AllowPrerelease
+                        $modulesWithPrelease = Find-Module -Name $pxModule.Name -AllowPrerelease -AllVersions
+                        $moduleToInstall = FindModuleFromList -Modules $modulesWithPrelease -ModuleToFind $pxModule -AllowPrerelease
+                    }
+                    else
+                    {
+                        if( -not $modules )
+                        {
+                            Write-Error 'No modules were found using the module names from the pxget file!' -ErrorAction SilentlyContinue
+                            continue
+                        }
+                        $moduleToInstall = FindModuleFromList -Modules $modules -ModuleToFind $pxModule
+                    }
+        
+                    if( -not $moduleToInstall )
+                    {
+                        Write-Error "Module $($pxModule.Name) was not found!"
+                        continue
+                    }
+        
+                    $installedModule =
+                        Get-Module -Name $pxModule.Name -List |
+                        Where-Object 'Version' -eq $moduleToInstall.Version
+                    # The latest version that matches the version in the pxget.json file is already installed
+                    if( $installedModule )
+                    {
+                        continue
+                    }
+        
+                    # Not installed. Install it. We pipe it so the repository of the module is also used.
+                    $moduleToInstall | Save-Module -Path $psmodulesPath

```

Comment:
After calling `Save-Module`, call `Get-Module` to return the module that got installed. This will be a nice indication to the user that pxget is doing something.

```powershell
$savedToPath = Join-Path -Path $psmodulesPath -ChildPath $moduleToInstall.Name
$savedToPath = Join-Path -Path $psmodulesPath -ChildPath ($moduleToInstall.Version -replace '-.*$', '')
Get-Module -Name $savedToPath -ListAvailable
```